Firestore/core/src/model/value_util.cc

```diff
@@ -509,51 +509,56 @@ Message<google_firestore_v1_Value> GetLowerBound(pb_size_t value_tag) {
     }
 
     case google_firestore_v1_Value_timestamp_value_tag: {
-      Message<google_firestore_v1_Value> value;
-      value->which_value_type = value_tag;
-      value->timestamp_value.seconds = std::numeric_limits<int64_t>::min();
-      value->timestamp_value.nanos = 0;
+      google_firestore_v1_Value value;
+      value.which_value_type = value_tag;
+      value.timestamp_value.seconds = std::numeric_limits<int64_t>::min();
+      value.timestamp_value.nanos = 0;
       return value;
     }
 
     case google_firestore_v1_Value_string_value_tag: {
-      Message<google_firestore_v1_Value> value;
-      value->which_value_type = value_tag;
-      value->string_value = nullptr;
+      google_firestore_v1_Value value;
+      value.which_value_type = value_tag;
+      value.string_value = nullptr;
       return value;
     }
 
     case google_firestore_v1_Value_bytes_value_tag: {
-      Message<google_firestore_v1_Value> value;
-      value->which_value_type = value_tag;
-      value->bytes_value = nullptr;
+      google_firestore_v1_Value value;
+      value.which_value_type = value_tag;
+      value.bytes_value = nullptr;
       return value;
     }
 
-    case google_firestore_v1_Value_reference_value_tag:
-      return RefValue(DatabaseId("", ""), DocumentKey::Empty());
+    case google_firestore_v1_Value_reference_value_tag: {
+      google_firestore_v1_Value result;
+      result.which_value_type = google_firestore_v1_Value_reference_value_tag;
+      result.reference_value =
```

Comment:
This is now leaking memory. Please use a static variable.

Reply:
Fixed.